properties/basic.py

```diff
@@ -754,11 +757,26 @@ def unicode(self, value):
             raise TypeError("'unicode' property must be a boolean")
         self._unicode = value
 
+    @property
+    def regexp(self):
+        """Regular expression the string must comply with"""
+        return getattr(self, '_regexp', None)
+
+    @regexp.setter
+    def regexp(self, value):
+        if not isinstance(value, string_types):
+            raise TypeError("'regexp' must be a string")
+        self._regexp = value
+
     def validate(self, instance, value):
         """Check if value is a string, and strips it and changes case"""
         value_type = type(value)
         if not isinstance(value, string_types):
             self.error(instance, value)
+        if self.regexp is None:
+            pass
+        elif re.match(self.regexp, value) is None:
+            self.error(instance, value)
```

Comment:
Do we want to pre-compile (or, at least, cache) this on the property? I'm thinking for cases when many instances are created that might speed it up. I.e., `re.compile`. I can put something together for your thoughts if you want.